src/main/java/ch/fhnw/algd1/converters/binary/BinConverter.java

```diff
@@ -2,14 +2,22 @@
 
 public class BinConverter {
 	public static String toString(int x) {
-		// TODO expect x to be in range [-128, 127], return string with 8 binary
-		// digits representing x in 2-complement
-		return "00000000";
+		StringBuilder binary = new StringBuilder();
+
+		for (int i = 7; i >= 0; i--) {
+			binary.append((x >> i) & 1);
```

Comment:
Die `append`-Methode, die hier aufgerufen wird, hat einen `int`-Parameter und führt deswegen ein Programm aus, dass eine beliebige `int`-Zahl in einen String umwandelt. Das ist eigentlich zu mächtig. Sie könnten diese "Umwandlung" für den speziellen Fall hier auch gerade selbst programmieren:
```suggestion
			binary.append(((x >> i) & 1) == 0 ? '0' : '1');
```
(oder so wie unten mit dem Wissen aus dem Lernmaterial der aktuellen Woche...)